buffer.go

```diff
@@ -3,50 +3,49 @@ package extsort
 import (
 	"container/heap"
 	"sort"
-
-	"github.com/valyala/bytebufferpool"
 )
 
-var bufferPool bytebufferpool.Pool
-
 type memBuffer struct {
-	size   int
-	chunks [][]byte
-	less   Less
+	size int
+	ents []entry
+	less Less
 }
 
-func (b *memBuffer) Append(data []byte) {
-	n := len(b.chunks)
-	if n < cap(b.chunks) {
-		b.chunks = b.chunks[:n+1]
+func (b *memBuffer) Append(key, val []byte) {
+	n := len(b.ents)
+	if n < cap(b.ents) {
+		b.ents = b.ents[:n+1]
```

Comment:
Does this make it perform observably faster? I see it's an old code adapted, but I'd better go with normal `append`.

Reply:
it is to reuse existing `entry` buffers

Reply:
basically, if there is an `entry` already stored at position `n+1` it will have an `entry.data` that we can reuse too